homeassistant/components/lektrico/binary_sensor.py

```diff
@@ -0,0 +1,118 @@
+from .const import DOMAIN
+
+
+@dataclass
+class LektricoBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """A class that describes the Lektrico binary sensor entities."""
+
+    @classmethod
+    def get_is_on(cls, data: Any) -> bool | None:
+        """Return None."""
+        return None
+
+
+@dataclass
+class HeadlessBinarySensorEntityDescription(LektricoBinarySensorEntityDescription):
+    """A class that describes the Lektrico No Authentication binary sensor entity."""
+
+    @classmethod
+    def get_is_on(cls, data: Any) -> bool:
+        """Get the headless."""
+        return bool(data.headless)
+
+
+SENSORS: tuple[LektricoBinarySensorEntityDescription, ...] = (
+    HeadlessBinarySensorEntityDescription(
+        key="headless",
+        name="No Authentication",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Lektrico charger based on a config entry."""
+    _lektrico_device: LektricoDeviceDataUpdateCoordinator = hass.data[DOMAIN][
+        entry.entry_id
+    ]
+    sensors = [
+        LektricoBinarySensor(
+            sensor_desc,
+            _lektrico_device,
+            entry.data[CONF_FRIENDLY_NAME],
+        )
+        for sensor_desc in SENSORS
+    ]
+
+    async_add_entities(sensors, False)
+
+
+class LektricoBinarySensor(CoordinatorEntity, BinarySensorEntity):
+    """The entity class for Lektrico charging stations binary sensors."""
+
+    entity_description: LektricoBinarySensorEntityDescription
+
+    def __init__(
+        self,
+        description: LektricoBinarySensorEntityDescription,
+        _lektrico_device: LektricoDeviceDataUpdateCoordinator,
+        friendly_name: str,
+    ) -> None:
+        """Initialize Lektrico charger."""
+        super().__init__(_lektrico_device)
+        self.friendly_name = friendly_name
+        self.serial_number = _lektrico_device.serial_number
+        self.board_revision = _lektrico_device.board_revision
+        self.entity_description = description
+
+        self._attr_name = f"{self.friendly_name} {description.name}"
+        self._attr_unique_id = f"{self.serial_number}_{description.name}"
+        # ex: 500006_No Authorisation
+
+        self._lektrico_device = _lektrico_device
+
+    @property
+    def is_on(self) -> bool | None:
+        """Return true if the binary sensor is on."""
+        return self.entity_description.get_is_on(self._lektrico_device.data)
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return device information about this Lektrico charger."""
+        return {
+            ATTR_IDENTIFIERS: {(DOMAIN, self.serial_number)},
+            ATTR_NAME: self.friendly_name,
+            ATTR_MANUFACTURER: "Lektrico",
+            ATTR_MODEL: f"1P7K {self.serial_number} rev.{self.board_revision}",
+            ATTR_SW_VERSION: self._lektrico_device.data.fw_version,
+        }
```

Comment:
Set `self._attr_device_info = DeviceInfo(.....)` in the constructor instead. It will allow you to remove this whole method.

Reply:
I made the changes. Thanks.